docs/DeveloperGuide.md

```diff
@@ -1107,13 +1107,253 @@ testers are expected to do more *exploratory* testing.
+
+2. Adding an order with invalid parameters
+
+   1. Prerequisites: Only 1 customer exist in the customer list.
+
+   2. Test case: `addo 0 n/Banana Roll` <br />
+     Expected: No order is added. Information panel remains the same. Error details shown in the status message.
+
+   3. Test case: `addo 2 n/Banana Roll` <br />
+     Expected: Similar to previous.
+
+### Listing orders
+
+1. Listing all orders
+
+   1. Prerequisites: Multiple orders in the orders list.
+
+   2. Test case: `listo`<br />
+     Expected: All orders are shown in the list. Success message shown in the status message.
+
+2. Listing orders with valid sorting and filter option
+
+   1. Prerequisites: Multiple orders in the orders list.
+
+   2. Test case: `listo s/status`<br>
+      Expected: All orders are shown in the list sorted by status. Success message shown in the status message.
+
+   3. Test case: `listo f/pending`<br>
+      Expected: Only orders whose status are 'pending' are shown in the list. Success message shown in the status message.
+
+   4. Test case: `listo s/name f/paid`<br>
+      Expected: Only orders whose status are 'paid' are shown in the list sorted by name. Success message shown in the status message.
+
+   5. Other valid test cases to try: `listo s/x`, `listo f/y`, `listo s/x f/y`, `...`
+      (where x is a valid sorting option and y is a valid filter option)<br>
+      Expected: List is sorted or filtered according to x and y.
+
+3. Listing orders with invalid sorting or filter options
+
+    1. Prerequisites: Multiple orders in the list.
+
+    2. Test case: `listo s/invalid f/pending`<br>
+       Expected: List remains the same. Error details shown in the status message.
+
+    3. Test case: `listo s/status f/invalid`<br>
+       Expected: List remains the same. Error details shown in the status message.
+
+    4. Other incorrect test cases to try: `listo s/x`, `listo f/y`, `listo s/x f/y`, `...`
+       (where x is an invalid sorting option and y is an invalid filter option)<br>
+       Expected: Similar to previous.
+
+### Finding orders
+
+1. Finding existing orders
+
+    1. Prerequisite: Only the orders 'Banana Cake' and 'Chocolate Roll' exists in the order list.
+   
+    2. Test case: `findo Banana`<br>
+       Expected: Only the order 'Banana Cake' is shown. Success message with number of listed orders (1) shown in the status message.
+   
+    3. Test case: `findo Cake Banana`<br>
+       Expected: Only the order 'Banana Cake' is shown. Success message with number of listed orders (1) shown in the status message.
+   
+    4. Test case: `findo Banana Chocolate`<br>
+       Expected: Both orders 'Banana Cake' and 'Chocolate Roll' are shown. Success message with number of listed orders (2) shown in the status message.
+   
+2. Finding non-existing orders
```

Comment:
should this be non-existent? Or would you want to use non-existing to standardise with the above

Reply:
Resolved!